directory/datadog_checks/directory/directory.py

```diff
@@ -54,15 +55,13 @@ def check(self, instance):
         custom_tags = instance.get('tags', [])
 
         if not exists(abs_directory):
-            if ignore_missing:
-                self.log.info(
-                    'DirectoryCheck: the directory `{}` does not exist. Skipping.'.format(abs_directory)
-                )
-                return
-
-            raise Exception(
-                'DirectoryCheck: the directory `{}` does not exist. Skipping.'.format(abs_directory)
-            )
+            msg = "Either directory '{}' doesn't exist or the Agent doesn't "\
+                  "have permissions to access it, skipping.".format(abs_directory)
+
+            if not ignore_missing:
+                raise ConfigurationError(msg)
+
+            self.log.info(msg)
```

Comment:
`info` or `warning`?

Reply:
You're right, let's raise the log level at this point since permission errors will be eventually possible